modules/thermal_hydraulics/src/materials/ADWallHTCGnielinskiAnnularMaterial.C

```diff
@@ -0,0 +1,135 @@
+  params.addParam<MaterialPropertyName>(
+      "p", FlowModelSinglePhase::PRESSURE, "Fluid pressure material property");
+  params.addParam<MaterialPropertyName>(
+      "T", FlowModelSinglePhase::TEMPERATURE, "Fluid temperature material property");
+  params.addParam<MaterialPropertyName>(
+      "T_wall", FlowModel::TEMPERATURE_WALL, "Wall temperature material property");
+  params.addRequiredParam<Real>("D_inner", "Inner diameter of the annulus [m]");
+  params.addRequiredParam<Real>("D_outer", "Outer diameter of the annulus [m]");
+  params.addRequiredParam<Real>("channel_length", "Channel length [m]");
+  params.addRequiredParam<bool>("at_inner_wall", "True if heat transfer is at inner wall");
+  params.addRequiredParam<bool>("fluid_is_gas", "True if the fluid is a gas");
+  params.addParam<Real>("gas_heating_correction_exponent",
+                        0,
+                        "Exponent for the ratio of bulk fluid temperature to wall temperature for "
+                        "the Nusselt number correction factor when heating a gas");
+  params.addRequiredParam<UserObjectName>("fluid_properties", "Fluid properties object");
+
+  params.addClassDescription("Computes wall heat transfer coefficient for gases and water in an "
+                             "annular flow channel using the Gnielinski correlation");
+
+  return params;
+}
+
+ADWallHTCGnielinskiAnnularMaterial::ADWallHTCGnielinskiAnnularMaterial(
+    const InputParameters & parameters)
+  : Material(parameters),
+    _htc_wall(declareADProperty<Real>("htc_wall")),
+    _rho(getADMaterialProperty<Real>("rho")),
+    _vel(getADMaterialProperty<Real>("vel")),
+    _k(getADMaterialProperty<Real>("k")),
+    _mu(getADMaterialProperty<Real>("mu")),
+    _cp(getADMaterialProperty<Real>("cp")),
+    _p(getADMaterialProperty<Real>("p")),
+    _T(getADMaterialProperty<Real>("T")),
+    _T_wall(getADMaterialProperty<Real>("T_wall")),
+    _D_inner(getParam<Real>("D_inner")),
+    _D_outer(getParam<Real>("D_outer")),
+    _D_h(_D_outer - _D_inner),
+    _a(_D_inner / _D_outer),
+    _L(getParam<Real>("channel_length")),
+    _at_inner_wall(getParam<bool>("at_inner_wall")),
+    _fluid_is_gas(getParam<bool>("fluid_is_gas")),
+    _n(getParam<Real>("gas_heating_correction_exponent")),
+    _provided_gas_heating_correction_exponent(isParamSetByUser("gas_heating_correction_exponent")),
+    _fp(getUserObject<SinglePhaseFluidProperties>("fluid_properties"))
+{
+}
+
+void
+ADWallHTCGnielinskiAnnularMaterial::computeQpProperties()
+{
+  const ADReal Pr = THM::Prandtl(_cp[_qp], _mu[_qp], _k[_qp]);
+
+  ADReal K;
+  if (_fluid_is_gas)
+  {
+    if (_T_wall[_qp] > _T[_qp] && !_provided_gas_heating_correction_exponent)
+      mooseError(
+          "If wall temperature ever exceeds the fluid temperature, even in iteration, and the "
+          "fluid is a gas, then the parameter 'gas_heating_correction_exponent' must be provided.");
```

Comment:
Isn't there a default value that could be used here?

Reply:
There's not really a good default. I'd rather them be aware of the importance of this exponent than just use a bad default without knowing.